client/src/livingdocs-component-app/app.js

```diff
@@ -0,0 +1,314 @@
+            comparison: '='
+          }
+        ]
+      },
+      displayOptions: this.selectedItem.toolRuntimeConfig.displayOptions
+    };
+
+    return qEnv.QServerBaseUrl
+      .then(QServerBaseUrl => {
+        if (this.selectedItem) {
+          return fetch(`${QServerBaseUrl}/rendering-info/${this.selectedItem.id}/${this.target.key}?toolRuntimeConfig=${encodeURI(JSON.stringify(toolRuntimeConfig))}`);
+        }
+        return {};
+      })
+      .then(res => {
+        if (res.ok && res.status >= 200 && res.status < 400) {
+          return res.json();
+        }
+        throw res.statusText;
+      })
+      .then(renderingInfo => {
+        // add stylesheets for target preview if any
+        if (this.target.preview && this.target.preview.stylesheets) {
+          if (!renderingInfo.stylesheets) {
+            renderingInfo.stylesheets = [];
+          }
+          this.target.preview.stylesheets.forEach(stylesheet => {
+            renderingInfo.stylesheets.push(stylesheet);
+          });
+        }
+
+        // add scripts for target preview if any
+        if (this.target.preview && this.target.preview.scripts) {
+          if (!renderingInfo.scripts) {
+            renderingInfo.scripts = [];
+          }
+          this.target.preview.scripts.forEach(script => {
+            renderingInfo.scripts.push(script);
+          });
+        }
+        return renderingInfo;
+      });
+  }
+
+  loadPreview() {
+    this.fetchRenderingInfo()
+      .then(renderingInfo => {
+        this.errorMessage = undefined;
+        this.renderingInfo = renderingInfo;
+      })
+      .catch(errorMessage => {
+        this.errorMessage = errorMessage;
+        this.renderingInfo = {};
+      });
+  }
+
+  loadDisplayOptions() {
+    let tool = '';
+    try {
+      tool = this.items
+      .filter(item => {
+        return item.conf._id === this.selectedItem.id;
+      })
+      .map(selected => {
+        return selected.conf.tool;
+      })[0];
+    } catch (e) {
+      throw new Error('no tool defined');
+    }
+
+    qEnv.QServerBaseUrl
+      .then(QServerBaseUrl => {
+        return fetch(`${QServerBaseUrl}/tools/${tool}/display-options-schema.json`);
+      })
+      .then(response => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw response;
+      })
+      .then(schema => {
+        if (schema.hasOwnProperty('properties')) {
+          // we just accept one object containing simple boolean type properties
+          Object.keys(schema.properties).forEach(propertyName => {
+            // maybe delete all non boolean properties here??
+            let displayOption = schema.properties[propertyName];
+            if (displayOption.title) {
+              schema.properties[propertyName].title = this.i18n.tr(`${tool}:${displayOption.title}`);
+            }
+          });
+        }
+        this.optionKeys = Object.keys(schema.properties);
```

Comment:
You do not need this. If you want to walk all object properties in a viewmodel use `for...in`(https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/for...in). If you need to walk them in a view, there is the `keys` value converter so you can do something like this `repeat.for="optionKey of options | keys"`